x/oracle/keeper/keeper.go

```diff
@@ -215,13 +226,12 @@ func (k Keeper) ProcessSignProphecy(ctx sdk.Context, networkDescriptor types.Net
 		return err
 	}
 
+	// update witness's lock burn nonce
+	k.SetWitnessLockBurnNonce(ctx, networkDescriptor, valAddr, prophecyInfo.GlobalNonce)
+
 	// emit the event when status from pending to success
+	// TODO - what happens here if we don't match StatusText_STATUS_TEXT_PENDING and StatusText_STATUS_TEXT_SUCCESS?
```

Comment:
@juniuszhou @banshee I think we should make it crystal clear what happens in every case:
- old = unspecified, new = pending (+fail/success noops)
- old = pending, new = failure
- old = pending, new = success

Reply:
As the comment explains, the EventTypeProphecyCompleted just emitted once, and only happened if the status changed from pending to success. will add comment in code.
old = unspecified, new = pending  the prophecy just created, not emit the event
old = unspecified, new = success no such path
old = pending, new = failure no failed status for prophecy anymore
old = pending, new = success the only case we will emit the event
old = success, new = success not emit the same event twice